relayer/workers/beefyrelayer/beefy-relaychain-listener.go

```diff
@@ -12,27 +9,30 @@ import (
 	"github.com/snowfork/go-substrate-rpc-client/v2/types"
 	"golang.org/x/sync/errgroup"
 
-	"github.com/snowfork/polkadot-ethereum/relayer/store"
+	"github.com/snowfork/polkadot-ethereum/relayer/chain/relaychain"
 	"github.com/snowfork/polkadot-ethereum/relayer/substrate"
+	"github.com/snowfork/polkadot-ethereum/relayer/workers/beefyrelayer/store"
 )
 
-type Listener struct {
-	config        *Config
-	conn          *Connection
-	beefyMessages chan<- store.BeefyRelayInfo
-	log           *logrus.Entry
+type BeefyRelaychainListener struct {
+	relaychainConfig *relaychain.Config
+	relaychainConn   *relaychain.Connection
+	beefyMessages    chan<- store.BeefyRelayInfo
+	log              *logrus.Entry
 }
 
-func NewListener(config *Config, conn *Connection, beefyMessages chan<- store.BeefyRelayInfo, log *logrus.Entry) *Listener {
-	return &Listener{
-		config:        config,
-		conn:          conn,
-		beefyMessages: beefyMessages,
-		log:           log,
+func NewBeefyRelaychainListener(relaychainConfig *relaychain.Config,
+	relaychainConn *relaychain.Connection, beefyMessages chan<- store.BeefyRelayInfo,
+	log *logrus.Entry) *BeefyRelaychainListener {
+	return &BeefyRelaychainListener{
+		relaychainConfig: relaychainConfig,
+		relaychainConn:   relaychainConn,
+		beefyMessages:    beefyMessages,
+		log:              log,
 	}
 }
 
-func (li *Listener) Start(ctx context.Context, eg *errgroup.Group) error {
+func (li *BeefyRelaychainListener) Start(ctx context.Context, eg *errgroup.Group) error {
```

Comment:
ditto: No need to rename this to `BeefyRelaychainListener`.

Reply:
i think in this context, the name is still valuable as there are multiple listeners with different purposes/sources